contrib/NeonVM/cell.c

```diff
@@ -217,39 +229,18 @@ Cell *cell_arrayIndexForWrite(Cell *c, size_t i)
         if (c->array == NULL) {
             fatal_error("Unable to reallcoate memory for write array.");
         }
-        if (c->array_size) {
-            for (n = c->array_size; n < i; n++) {
-                cell_resetCell(&c->array[n]);
-            }
-            c->array->array_size = i;
-        } else {
-            c->array_size++;
-        }
+        c->array_size = i+1;
+        cell_resetCell(&c->array[i]);
```

Comment:
You probably want to initialize all the new cells that you just allocated memory for. If you don't, it's just junk memory with random bytes.